### PR TITLE
Store speech priorities in an enum

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Briefly discussed in #10371 

### Summary of the issue:
Speech refactor introduced speech priorities, which were constants.

### Description of how this pull request fixes the issue:
As Python 3 has an Enum type and this code is new to the Python 3 version of NVDA, use an enum instead.

This allso adds some type hints to speech functions.

### Testing performed:
Tested running from source. Tested that speech priorities still work, i.e. that higher prio speech interrupts lower prio speech.

### Known issues with pull request:
None

### Change log entry:
None needed.

### CODE REVIEW

**Comment by feerrenrut** [comment]
It is probably worth adding a clear note in the description of this PR to addon developers and posting in the mailing lists about this change.

File: source/speech/priorities.py [hunk 1]
```diff
@@ -1,19 +1,27 @@
 # -*- coding: UTF-8 -*-
-#A part of NonVisual Desktop Access (NVDA)
-#This file is covered by the GNU General Public License.
-#See the file COPYING for more details.
-#Copyright (C) 2006-2019 NV Access Limited
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2017-2019 NV Access Limited, Babbage B.V.
```
Comment (feerrenrut) [inline source/speech/priorities.py]: I think this should stay as 2006-2019
```suggestion
# Copyright (C) 2006-2019 NV Access Limited, Babbage B.V.
```

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Thanks @leonardder

File: source/speech/priorities.py (hunk 1 above)
Comment (leonardder) [reply source/speech/priorities.py]: I made this 2017 on purpose. @jcsteh started working on speech refactor somewhere around that year, right? Speech priorities are part of speech refactor, so they weren't there in 2006.

**Review by feerrenrut — APPROVED** [review]
Ok, thanks @leonardder

**Pull request merged**